File: morpion/src/modele/morpionSimple.java
package modele;

public class morpionSimple extends Morpion {
    public morpionSimple(int taille){
        super(taille);
    }
    public boolean aGagne(String symbole, int y, int x){
        String aTrouver=symbole.repeat(3);
//        Ligne
        String ligne="";
        for (String li : this.grille[y]){
            ligne+=li;
        }
        if(ligne.indexOf(aTrouver)>=0){
            return true;
        }
//        Colonne
        String colonne="";
        for (String[] col : this.grille){
            colonne+=col[x];
        }
        if (colonne.indexOf(aTrouver)>=0){
            return true;
        }
//        Diagonale
        if(x==y){
            String diagonale="";
            for(int lc=0;lc<this.taille;lc++){
                diagonale+=this.grille[lc][lc];
            }
            if(diagonale.indexOf(aTrouver)>=0){
                return true;
            }
        }
//        diagonale inverse
        if(x==this.taille-(y-1)){
            String diaginverse="";
            for(int lcinverse=0;lcinverse<this.taille;lcinverse++){
                diaginverse+=this.grille[lcinverse][this.taille-(lcinverse+1)];
            }
            if (diaginverse.indexOf(aTrouver)>=0){
                return true;
            }
        }
        return false;
    }

}

File: morpion/js/morpionSimple.ts
import {Morpion} from './morpion.js';

export class MorpionSimple extends Morpion {

    aGagne(symbole:string, y:number, x:number) {
        let aTrouver:string = symbole.repeat(3);

        // gagné en ligne ? : concaténation de la ligne, et recherche de la sous-chaîne gagnante
        let ligne:string = '';
        this.grille[y].forEach(element => ligne += element);
        if (ligne.indexOf(aTrouver) >= 0) {
            return true;
        }

        // gagné en colonne ? : concaténation de la colonne et recherche de la sous-chaîne gagnante
        let col:string = '';
        this.grille.forEach(element => col += element[x]);
        if (col.indexOf(aTrouver) >= 0) {
            return true;
        }

        // gagné diagonale
        if (x === y) {
            let diagonale:string = '';
            for (let lc:number = 0; lc < this.taille; lc++) {
                diagonale += this.grille[lc][lc];
            }
            if (diagonale.indexOf(aTrouver) >= 0) {
                return true;
            }
        }

        // gagné diag inverse
        if (x === this.taille - (y + 1)) {
            let inverse:string = '';
            for (let lc:number = 0; lc < this.taille; lc++) {
                inverse += this.grille[lc][this.taille - (lc + 1)];
            }
            if (inverse.indexOf(aTrouver) >= 0) {
                return true;
            }
        }

        return false;

    }
}
